common/fee/utils.go

```diff
@@ -3,21 +3,20 @@ package fee
 import (
 	"math/big"
 
-	bigmath "github.com/smartcontractkit/chainlink/v2/core/utils/big_math"
-
 	"github.com/shopspring/decimal"
 )
 
-func GetMaxFeePrice(userSpecifiedMax, maxFee *big.Int) *big.Int {
-	return bigmath.Min(userSpecifiedMax, maxFee)
+func ApplyMultiplier(feeLimit uint32, multiplier float32) uint32 {
+	return uint32(decimal.NewFromBigInt(big.NewInt(0).SetUint64(uint64(feeLimit)), 0).Mul(decimal.NewFromFloat32(multiplier)).IntPart())
```

Comment:
just wondering: would we ever be concerned this overflows?

my guess is unlikely since this is a multiplier on the individual fee units (gas, compute units) used but technically not impossible if a chain decides to have a huge block size

Reply:
i think thats a good point. 

If the product exceeds 4,294,967,295, then there would be an integer overflow when we cast it back to uint32. 

Should feeLimit be big.Int instead of uint32 then? I think the minimum would be to add checks and error handling if its more than that amount. 

Wondering if across chain, the limit would be more than `4,294,967,295` 🤔